firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/PatchMutation.java

```diff
@@ -142,6 +143,32 @@ public void applyToLocalView(MutableDocument document, Timestamp localWriteTime)
         .setHasLocalMutations();
   }
 
+  @Override
+  public Mutation squash(
+      Mutation baseMutation, MutableDocument document, Timestamp localWriteTime) {
+    if (!getPrecondition().isValidFor(document)) {
```

Comment:
precondition should be tested against `document` + `baseMutation`, isn't it?

Reply:
I don't think so (but I could be wrong). `baseMutation` is a terrible name (as we are already using it somewhere else in this client with a different meaning), but this `baseMutation` should already have passed all preconditions, otherwise it would not have changed during the last call to `squash`.